### PR TITLE
Extended KDNode with a public method returning the node's splitting axis

### PR DESCRIPTION
**Pull request by wsmigaj**
Addition of this method will make it possible to write custom visitors traversing a kd-tree and performing spatial queries different from the ones supported natively by the ``KDNode`` class (i.e. nearest neighbour search and retrieval of points lying in a sphere).  

I'd like to use it for queries checking whether any point in a kd-tree intersects an ellipsoid or a cylinder (https://github.com/JCSDA/oops/pull/497).

### CODE REVIEW

File: src/eckit/container/kdtree/KDNode.h [hunk 1]
```diff
@@ -45,6 +45,8 @@ class KDNode : public SPNode<Traits, KDNode<Traits> > {
 
     static KDNode<Traits>* insert(Alloc& a, const Value& value, KDNode<Traits>* node, int depth = 0);
 
+    /// Return the axis along which this node is split.
+    size_t axis() const { return axis_; }
```
Comment (wsmigaj) [inline src/eckit/container/kdtree/KDNode.h]: The parent ``SPNode`` class already gives clients read-only access to its point, payload and children nodes, so I think making the splitting axis publicly accessible too is in line with the rest of the interface.

**Comment by wsmigaj** [comment]
@mmiesch This is the PR I mentioned during today's meeting. It would be great if it could be reviewed before ``eckit`` is updated in the containers, especially since it's only a small change.

**Comment by mmiesch** [comment]
Yep - I agree too - I just want to run a quick test before I approve.

**Comment by mmiesch** [comment]
@wsmigaj - would you mind creating this and future PRs as a branch in the jcsda repository rather than your own fork?  It would make it easier to test.

File: src/eckit/container/kdtree/KDNode.h (hunk 1 above)
Comment (mmiesch) [reply src/eckit/container/kdtree/KDNode.h]: I ran all the tests in eckit, fckit, atlas, and ufo-bundle just to make sure this didn't do anything unexpected but all looks good.

**Comment by wsmigaj** [comment]
Many thanks for the rapid merge!

@markjolah I've opened a PR in the upstream repository: https://github.com/ecmwf/eckit/pull/19.

@mmiesch The reason why I've created this PR in my fork of ``eckit`` is that I don't seem to be able to push to ``JCSDA/eckit``: when I try, I get the error
```
remote: Permission to JCSDA/eckit.git denied to wsmigaj.
fatal: unable to access 'https://github.com/JCSDA/eckit.git/': The requested URL returned error: 403
```

**Pull request merged**